packages/meditrak-server/src/tests/routes/GETDashboardGroups/assertDashboardGroupsPermissions.test.js

```diff
@@ -132,214 +130,98 @@ describe('Permissions checker for GETDashboardGroups', async () => {
+  describe('GET /dashboardGroups', async () => {
+    it('Sufficient permissions: Return only the list of entries we have permissions for', async () => {
       const policy = {
         DL: ['Public'],
-        KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, /*'Admin'*/ 'Public'],
+        KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, /* 'Admin', */ 'Public'],
         SB: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Royal Australasian College of Surgeons'],
         VU: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin'],
-        LA: ['Admin', 'Public'],
+        LA: ['Admin'],
         TO: ['Admin'],
       };
-      const accessPolicy = new AccessPolicy(policy);
+      await prepareStubAndAuthenticate(app, policy);
+      const { body: results } = await app.get(`dashboardGroups`);
 
-      expect(() =>
-        assertDashboardGroupsPermissions(accessPolicy, models, [
-          facilityDashboardGroup1,
-          nationalDashboardGroup2,
-        ]),
-      ).to.throw;
+      expect(results.map(r => r.id)).to.deep.equal([
+        nationalDashboardGroup2.id,
+        projectLevelDashboardGroup.id,
+      ]);
     });
 
-    it('Insufficient permissions: Should throw an exception if users do not have access to any of the dashboard groups', async () => {
-      //Remove the Admin permission of LA to have insufficient permissions to access nationalDashboardGroup2.
-      const policy = {
-        DL: ['Public'],
-        KI: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin'],
-        SB: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Royal Australasian College of Surgeons'],
-        VU: [TUPAIA_ADMIN_PANEL_PERMISSION_GROUP, 'Admin'],
-        LA: [/*'Admin'*/ 'Public'],
-        TO: ['Admin'],
-      };
-      const accessPolicy = new AccessPolicy(policy);
+    it('Sufficient permissions: Should return the full list of dashboard groups if we have BES Admin access', async () => {
+      await prepareStubAndAuthenticate(app, BES_ADMIN_POLICY);
+      const { body: results } = await app.get(`dashboardGroups`);
 
-      expect(() =>
-        assertDashboardGroupsPermissions(accessPolicy, models, [
-          nationalDashboardGroup1,
-          nationalDashboardGroup2,
-        ]),
-      ).to.throw;
+      expect(results.map(r => r.id)).to.deep.equal([
+        facilityDashboardGroup1.id,
+        districtDashboardGroup1.id,
+        nationalDashboardGroup1.id,
+        nationalDashboardGroup2.id,
+        projectLevelDashboardGroup.id,
+      ]);
     });
 
-    it('Insufficient permissions: Should throw an exception if dashboard groups are project level and users do not have access to any of their child countries', async () => {
-      //Remove Admin permission of TO, SB, KI, VU to have insufficient permissions to access the project level dashboard groups.
+    it('Insufficient permissions: Should throw an exception if we have permissions for no dashboard groups', async () => {
```

Comment:
I'm starting to question this as a general policy. What if the reason there are no results is that we specified an invalid search term, rather than a permissions issue? I guess here it does work, but we should either apply that on a case by case basis, or return an empty array if we want a general rule across endpoints. 

What do you think @EMcQ-BES?

Reply:
I guess this made more sense when we were running these as filters since we could know for certain the permissions step caused the empty array. I think returning an empty array is fair.

Do we still want to throw an error when requesting a single record we don't have permission for? We can tell for certain on a single record that it's a permissions error, but we may prefer to keep consistency between those responses.

Reply:
Let's keep the error on single record. They've specified an id, so it's quite clear that they're trying to do something they shouldn't be able to, as you say.